functions-methods.js

```diff
@@ -2,35 +2,93 @@
-// typeOfEmail("a.wiersma@outlook.com") geeft "Extern"
-
-
-
-/* Opdracht  3 */
-// Schrijf een functie genaamd checkEmailValidity, die een emailadres verwacht en checkt of het emailadres valide is. De functie returned true of false, afhankelijk van de uitkomst.
-// Een emailadres is valide wanneer:
-// * Er een @ in voorkomt
-// * Er géén , in voorkomt
-// * Er géén . in voorkomt als allerlaatste karakter (dus hotmail.com is valide, net als outlook.nl, maar outlooknl. niet)
-// ---- Verwachte uitkomsten:
-// checkEmailValidity("n.eeken@novi.nl") geeft true - want @ en punt op de juiste plek
-// checkEmailValidity("tessmellink@novi.nl") geeft true - want @ en punt op de juiste plek
-// checkEmailValidity("n.eekenanovi.nl") geeft false - want geen @
-// checkEmailValidity("n.eeken@novinl.") geeft false - want de punt mag niet als laatst
-// checkEmailValidity("tessmellink@novi,nl") geeft false - want er staat een komma in
\ No newline at end of file
+/* Opdracht 1
+    [v] Schrijf een functie genaamd getEmailDomain,
+    [v] die een emailadres verwacht en de domeinnaam teruggeeft.
+    [v] Een domeinnaam is hetgeen dat na het @ in het adres staat
+        ---- Verwachte uitkomsten:
+    [v] getEmailDomain("n.eeken@novi-education.nl") geeft novi-education.nl
+    [v] getEmailDomain("t.mellink@novi.nl") geeft novi.nl
+    [v] getEmailDomain("a.wiersma@outlook.com") geeft outlook.com
+*/
+
+console.log("================opdracht1======================");
+console.log(getEmailDomain("n.eeken@novi-education.nl"));
+console.log(getEmailDomain("t.mellink@novi.nl"));
+console.log(getEmailDomain("a.wiersma@outlook.com"));
+
+function getEmailDomain(emailAdres) {
+
+    let domeinnaam = [];
+
+    domeinnaam = emailAdres.split("@").reverse();
+
+    return domeinnaam[0];
+}
+
+/* Opdracht 2
+    [v] Schrijf een functie genaamd typeOfEmail, die een emailadres verwacht.
+    [v] De functie checkt of het emailadres een novi domein heeft (medewerker),
+    [v] een novi-education domein (student),
+    [v] of extern domein (zoals gmail of outlook)
+        ---- Verwachte uitkomsten:
+    [v] typeOfEmail("n.eeken@novi-education.nl") geeft "Student"
+    [v] typeOfEmail("t.mellink@novi.nl") geeft geeft "Medewerker"
+    [v] typeOfEmail("novi.nlaapjesk@outlook.com") geeft geeft "Extern" <-- deze moet het ook doen!
+    [v] typeOfEmail("a.wiersma@outlook.com") geeft "Extern"
+*/
+
+console.log("================opdracht2======================");
+console.log(typeOfEmail("n.eeken@novi-education.nl"));
+console.log(typeOfEmail("t.mellink@novi.nl"));
+console.log(typeOfEmail("novi.nlaapjesk@outlook.com"));
+console.log(typeOfEmail("a.wiersma@outlook.com"));
+
+function typeOfEmail(emailAdres) {
+
+    emailAdres = getEmailDomain(emailAdres)
+    let answer = "Extern";
+
+    if (emailAdres.includes("novi")) {
+        answer = "Medewerker";
+    }
+    if (emailAdres.includes("novi-education")) {
+        answer = "Student";
+    }
+    return answer;
```

Comment:
Ik zou er een if-else van maken en je zou de string meteen terug kunnen geven:
```suggestion
    if (emailAdres.includes("novi-education")) {
        return "Student";
    } else if (emailAdres.includes("novi")) {
        return "Medewerker";
    }
    
    return "Extern";
```